modules/collect/src/main/java/com/opengamma/collect/timeseries/LocalDateDoubleTimeSeries.java

```diff
@@ -545,21 +257,7 @@ public LocalDateDoubleTimeSeries mapValues(DoubleUnaryOperator mapper) {
    * @param predicate  the predicate to use to the filter the elements of the series
    * @return a filtered version of the series
    */
-  public LocalDateDoubleTimeSeries filter(ObjDoublePredicate<LocalDate> predicate) {
-    ArgChecker.notNull(predicate, "predicate");
-    // build up result in arrays keeping track of count of retained dates
-    LocalDate[] resDates = new LocalDate[size()];
-    double[] resValues = new double[size()];
-    int resCount = 0;
-    for (int i = 0; i < size(); i++) {
-      if (predicate.test(dates[i], values[i])) {
-        resDates[resCount] = dates[i];
-        resValues[resCount] = values[i];
-        resCount++;
-      }
-    }
-    return createUnsafe(Arrays.copyOf(resDates, resCount), Arrays.copyOf(resValues, resCount));
-  }
+  public abstract LocalDateDoubleTimeSeries filter(ObjDoublePredicate<LocalDate> predicate);
```

Comment:
I think it's still valid - for both sparse and dense you are going to iterate over all the values in the underlying arrays. Subseries will jump you to the correct subset more efficiently in both cases.